File: src/web/magmac/app/lang/node/PlantUMLRootSegments.ts
import { Lists } from "../../../../magmac/api/collect/list/Lists";
import { OrRule } from "../../../../magmac/app/compile/rule/OrRule";
import { Rule } from "../../../../magmac/app/compile/rule/Rule";
import { SuffixRule } from "../../../../magmac/app/compile/rule/SuffixRule";
export class PlantUMLRootSegments {
	createRootSegmentRule() : Rule {return new SuffixRule( new OrRule( 0.of( 0.createRule( ), 0.createRule( ), 0.createStructureRule( 0), 0.createStructureRule( 0), 0.createStructureRule( 0), 0.createInheritsRule( ), 0.createDependencyRule( ))), 0);;}
}

File: packages/compiler-java/src/main/java/magmac/app/lang/node/PlantUMLRootSegments.java
package magmac.app.lang.node;

/**
 * Builds the parser rules for all valid PlantUML root segments.
 */

import magmac.api.collect.list.Lists;
import magmac.app.compile.rule.OrRule;
import magmac.app.compile.rule.Rule;
import magmac.app.compile.rule.SuffixRule;

final class PlantUMLRootSegments {
    public static Rule createRootSegmentRule() {
        return new SuffixRule(new OrRule(Lists.of(
                PlantUMLHeader.createRule(),
                PlantUMLFooter.createRule(),
                PlantUMLStructure.createStructureRule("class"),
                PlantUMLStructure.createStructureRule("interface"),
                PlantUMLStructure.createStructureRule("enum"),
                PlantUMLInherits.createInheritsRule(),
                PlantUMLDependency.createDependencyRule()
        )), "\n");
    }
}
